src/Ast/ComparisonNode.php

```diff
@@ -4,73 +4,152 @@
 
 class ComparisonNode implements Node
 {
-    private const EQUAL     = 0;
-    private const NOT_EQUAL = 1;
-    private const IN        = 2;
-
-    private $left;
-    private $right;
+    public const EQUALS         = 'EQUALS';
+    public const GREATER        = 'GREATER';
+    public const GREATER_EQUALS = 'GREATER_EQUALS';
+    public const LESS           = 'LESS';
+    public const LESS_EQUALS    = 'LESS_EQUALS';
+    public const IN             = 'IN';
+    public const MATCHES        = 'MATCHES';
+
+    private $identifier;
+    private $value;
     private $type;
 
 
-    private function __construct(int $type, IdentifierNode $left, Node $right)
+    private function __construct(string $type, IdentifierNode $identifier, LiteralNode $value)
     {
-        $this->left  = $left;
-        $this->right = $right;
-        $this->type  = $type;
+        $this->identifier = $identifier;
+        $this->value      = $value;
+        $this->type       = $type;
     }
 
 
-    public function left() : IdentifierNode
+    public static function equals(IdentifierNode $left, LiteralNode $right) : self
     {
-        return $this->left;
+        return new static(self::EQUALS, $left, $right);
     }
 
 
-    public function right() : Node
+    public static function greaterThan(IdentifierNode $left, LiteralNode $right) : self
     {
-        return $this->right;
+        return new static(self::GREATER, $left, $right);
     }
 
 
-    public static function equal(IdentifierNode $left, Node $right) : self
+    public static function greaterThanOrEqualTo(IdentifierNode $left, LiteralNode $right) : self
     {
-        return new static(self::EQUAL, $left, $right);
+        return new static(self::GREATER_EQUALS, $left, $right);
     }
 
 
-    public static function notEqual(IdentifierNode $left, Node $right) : self
+    public static function lessThan(IdentifierNode $left, LiteralNode $right) : self
     {
-        return new static(self::NOT_EQUAL, $left, $right);
+        return new static(self::LESS, $left, $right);
     }
 
 
-    public static function in(IdentifierNode $left, NodeList $right) : self
+    public static function lessThanOrEqualTo(IdentifierNode $left, LiteralNode $right) : self
+    {
+        return new static(self::LESS_EQUALS, $left, $right);
+    }
+
+
+    public static function in(IdentifierNode $left, LiteralNodeList $right) : self
     {
         return new static(self::IN, $left, $right);
     }
 
 
+    public static function matches(IdentifierNode $left, StringNode $right) : self
+    {
+        return new static(self::MATCHES, $left, $right);
+    }
+
+
     public function accept(Visitor $visitor) : void
     {
         $visitor->visitComparison($this);
     }
 
 
-    public function isEqual() : bool
+    public function is(string $type) : bool
```

Comment:
With other methods like `isEquals`, `isGreaterThan` do we need this? It would be nicer to keep the type constants private imo.

Reply:
Yeah it was only used in tests, I've removed it and made the constants private.